lib/DocumentInspectorFactoryInterface.php

```diff
@@ -11,27 +11,29 @@
 
 namespace Sulu\Component\DocumentManager;
 
+use Sulu\Component\DocumentManager\DocumentManagerContext;
+
 /**
  * Document inspector factory.
  *
  * Allows the client to implement their own DocumentInspector implementations.
  *
- * As Document inspectors are dependent upon a document manager, and as the
- * number of document managers is unbounded it would be impractical to manually
- * set the document manager on each individual instance.
+ * As Document inspectors are dependent upon a document context, and as the
+ * number of document contexts is unbounded it would be impractical to manually
+ * set the document context on each individual instance.
  *
  * The factory should create the DocumentInspector instance using the given
- * document manager.
+ * document context.
  */
 interface DocumentInspectorFactoryInterface
 {
     /**
      * Create a new DocumentInspector instance using the given document
-     * manager.
+     * context.
      *
-     * @param DocumentManagerInterface
+     * @param DocumentManagerContext
      *
      * @return DocumentInspector
      */
-    public function getInspector(DocumentManagerInterface $manager);
+    public function getInspector(DocumentManagerContext $context);
```

Comment:
The problem for me is that it adds complexity and inconsistency. The same argument could apply for other parts of the context as well, and we don't introduce factories for all of them. Also, some of the classes are created in the `DocumentManager`'s constructor. For me that's very confusing, and takes some time to understand, and people won't remember it. If we put all of it in the `Extension`, there is a single place we know we have to look this kind of stuff up. And this functionality can also be tested in this bundle.